src/components/text_input/text_input.ts

```diff
@@ -64,11 +64,44 @@ export class TextInput extends Component<Props, SpreadsheetChildEnv> {
     );
   }
 
+  onKeyDown(ev: KeyboardEvent) {
+    switch (ev.key) {
+      case "Enter":
+        this.save();
+        this.inputRef.el!.blur();
+        ev.preventDefault();
+        ev.stopPropagation();
+        break;
+      case "Escape":
+        this.inputRef.el!.value = this.props.value;
+        this.inputRef.el!.blur();
```

Comment:
A bit the same here, `onBlur` will call `this.props.onChange`, that might call an `update_pivot` or whatever for nothing. Maybe the `save` could not do anything if the current content is the same as `props.content` ?